Docs: Add info about dependencies for local development (#1375)
grafana/plugin-tools: docusaurus/docs/get-started/set-up-development-environment.mdx

-"dev": "webpack -w -c ./.config/webpack/webpack.config.ts --env development",
+"dev": "webpack -w -c ./webpack.config.ts --env development",
```
## Add a dependency for local development

Grafana’s plugin development environment is designed to simplify the process of creating and testing plugins. Here’s how it operates:

Plugin distribution path:

> The plugin’s dist folder (built plugin) is linked to `/var/lib/grafana/plugins/{{pluginId}}` in the Grafana server environment.

Reviewer: I don't think is a good idea to be naming paths like this in the documentation because this can become very confusing quickly.

This text is true, but only if you are running grafana from a binary, in linux or mac. if you are running it with docker this is not longer valid because docker configuration often mount local folders, or if you clone and build the grafana repository it is also not the case. 

Better to link the official documentation https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/#plugins

Author: done.

Plugin initialization:

> When the Grafana server starts, its plugin loader runs a [Load](https://github.com/grafana/grafana/blob/b53f14ca83eeec6b0f624c2dbbaa417d08bbadbf/pkg/plugins/manager/loader/loader.go#L55-L56) function. This scans the `/var/lib/grafana/` plugins directory for plugins, validates `plugin.json` files, checks signatures and manifests, and initializes valid plugins.

Reviewer: again, let's not mention paths specifically here
```suggestion
When the Grafana server starts, its plugin loader runs a [Load](https://github.com/grafana/grafana/blob/b53f14ca83eeec6b0f624c2dbbaa417d08bbadbf/pkg/plugins/manager/loader/loader.go#L55-L56) function. This scans the defined [plugins directory](https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/#plugins) for plugins, validates `plugin.json` files, checks signatures and manifests, and initializes valid plugins.
```

Dependency management:

> Although some validations occur on the server side, the plugin path itself doesn't include dependency validation. Developers are responsible for ensuring dependencies are correctly managed.

Reviewer: I generally don't understand where this is going in terms of dependency management, maybe what we want to say here is that grafana doesn't `resolve` dependencies automatically?

Author: i've tweaked this paragraph to make it clearly refer to plugin dependencies.

To manage dependencies during local plugin development, you can use one of the following methods:

### Use the `GF_INSTALL_PLUGINS` environment variable

> * Add dependencies explicitly to your Grafana instance with `GF_INSTALL_PLUGINS=dependency`.

Reviewer: an example of this is any plugin we have in the catalog, so

`GF_INSTALL_PLUGINS=grafana-clock-panel`

tells grafana to install that plugin.

Author: added this example.

> Note that this approach is commonly used during development but requires manual tracking and updates for dependencies.

Reviewer: It is not clear what does `requires manual tracking and updates for dependencies.` mean. what is the intention in this text?

Author: Because it's not clear, I may have misunderstood something. Removing note.

> ### Build and install the plugin via the Grafana CLI

Reviewer: this section is not using grafana CLI at all

Author: removed erroneous phrase

-"dev": "webpack -w -c ./.config/webpack/webpack.config.ts --env development",
+"dev": "webpack -w -c ./webpack.config.ts --env development",
```

> ## Add a dependency for local development

Reviewer: ```suggestion
## Add a plugin as a dependency for local development
```

Grafana’s plugin development environment is designed to allow for plugins to be dependent upon other plugins if necessary. Here’s how it operates:

> Plugin distribution path:

Reviewer: ```suggestion
Plugins location directory:
```
